src/dev/run_check_published_api_changes.ts

```diff
@@ -259,20 +277,22 @@ async function run(
   }
 
   try {
-    log.info(`Core: Building types...`);
+    log.info(`Building types for api extractor...`);
     await runBuildTypes();
   } catch (e) {
     log.error(e);
     return false;
   }
 
-  const results = await Promise.all(
-    folders
-      .filter((folder) => (opts.filter.length ? folder.match(opts.filter) : true))
-      .map((folder) => run(folder, { log, opts }))
+  const filteredFolders = folders.filter((folder) =>
+    opts.filter.length ? folder.match(opts.filter) : true
   );
+  const results = [];
+  for (const folder of filteredFolders) {
+    results.push(await run(folder, { log, opts }));
```

Comment:
instead of running all in parallel now it runs seq. 
it is longer, but not crushes with OOM. (with plugin number grows, it would outgrow 10-16-32gb pretty fast)

Reply:
It would be a bit more work, but since this is CPU and memory heavy and we're adding more and more plugins, maybe you could execute a child process for each folder? https://nodejs.org/api/child_process.html#child_process_child_process_exec_command_options_callback

Reply:
I played around with node worker threads in https://github.com/elastic/kibana/pull/69615 Although it reduces the memory consumption (no need for extra heap) the speed up isn't enormous (1m50 vs 3m04s) so I think it's fine to leave this synchronous until we have Node12.

Reply:
That's awesome!